runtime/src/iree/io/parameter_index_provider.c

```diff
@@ -370,18 +423,28 @@ static iree_status_t iree_io_parameter_index_provider_write(
   IREE_RETURN_AND_END_ZONE_IF_ERROR(
       z0, iree_io_parameter_index_provider_resolve(
               provider, device, queue_affinity, target_scope, target_key,
-              IREE_HAL_MEMORY_ACCESS_READ, &target_entry, &target_file));
+              IREE_HAL_MEMORY_ACCESS_WRITE, &target_entry, &target_file));
```

Comment:
Is this just an unrelated bug you noticed and fixed?

Reply:
yep! we don't have anything exercising the write path today - would have been first thing hit when trying it but noticed it while reading